network_wrangler/transit/projects/add_route.py

```diff
@@ -89,71 +89,66 @@ def _add_route_to_feed(
     add_routes_df = pd.DataFrame(
         [{k: v for k, v in r.items() if k != "trips"} for r in add_routes]
     )
-    routes_df = pd.concat([feed.routes, add_routes_df], ignore_index=True, sort=False)
+    feed.routes = pd.concat([feed.routes, add_routes_df], ignore_index=True, sort=False)
 
     for route in add_routes:
         WranglerLogger.debug(
             f"Adding {len(route['trips'])} trips for route {route['route_id']} to feed."
         )
-        add_trips_df = _create_new_trips(route["trips"], route["route_id"], feed.shapes)
-        trips_df = pd.concat([feed.trips, add_trips_df], ignore_index=True, sort=False)
 
         for i, trip in enumerate(route["trips"]):
-            trip_id = add_trips_df.iloc[i]["trip_id"]
-            shape_id = add_trips_df.iloc[i]["shape_id"]
-            stop_dicts = _get_stops_from_routing(trip["routing"])
-
+            shape_id = fill_str_ids(pd.Series([None]), feed.shapes["shape_id"]).iloc[0]
             add_shape_df = _create_new_shape(trip["routing"], shape_id, road_net)
-            add_freqs_df = _create_new_frequencies(trip["headway_secs"], trip_id)
-            add_stop_times_df = _create_new_stop_times(stop_dicts, trip_id)
-            add_stops_df = _create_new_stops(
-                add_stop_times_df["stop_id"], feed.stops["stop_id"], road_net
-            )
-
-            # Add new data to existing dataframes
-            shapes_df = pd.concat([feed.shapes, add_shape_df], ignore_index=True, sort=False)
-            frequencies_df = pd.concat(
-                [feed.frequencies, add_freqs_df], ignore_index=True, sort=False
-            )
-            stops_df = pd.concat([feed.stops, add_stops_df], ignore_index=True, sort=False)
-            stop_times_df = pd.concat(
-                [feed.stop_times, add_stop_times_df], ignore_index=True, sort=False
-            )
-
-    feed.stops = stops_df
-    feed.routes = routes_df
-    feed.shapes = shapes_df
-    feed.trips = trips_df
-    feed.stop_times = stop_times_df
-    feed.frequencies = frequencies_df
+            if route['agency_raw_name']:
+                add_shape_df['agency_raw_name'] = route['agency_raw_name']
+            feed.shapes = pd.concat([feed.shapes, add_shape_df], ignore_index=True, sort=False)
+            
+            for j, headway in enumerate(trip["headway_secs"]):
+                trip_id = f"trip{j}_shp{shape_id}"
+                stop_dicts = _get_stops_from_routing(trip["routing"])
+                add_trips_df = _create_new_trips(trip, route["route_id"], trip_id, shape_id)
+                add_freqs_df = _create_new_frequencies(headway, trip_id)
+                add_stop_times_df = _create_new_stop_times(stop_dicts, trip_id)
+                add_stops_df = _create_new_stops(
+                    add_stop_times_df["stop_id"], feed.stops["stop_id"], road_net
+                )
+
+                if route['agency_raw_name']:
```

Comment:
I don't think we need this in our datamodel anymore.  Not sure what functionality it is providing?

Reply:
Cube/Metcouncil wrangler uses `agency_raw_name` to do some data queries. @i-am-sijia Do we need to keep it?

Reply:
you can keep it, but just either use generalized methods to perpetuate it into new routes or add it to cube/metcouncil logic